sdk/identity/Azure.Identity/src/ServiceFabricManagedIdentitySource.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Identity
+{
+    internal class ServiceFabricManagedIdentitySource : ManagedIdentitySource
+    {
+        private const string ServiceFabricMsiApiVersion = "2019-07-01-preview";
+        private const string IdentityEndpointInvalidUriError = "The environment variable IDENTITY_ENDPOINT contains an invalid Uri.";
+
+        private readonly Uri _endpoint;
+        private readonly string _identityHeaderValue;
+        private readonly string _clientId;
+
+        public static ManagedIdentitySource TryCreate(CredentialPipeline pipeline, string clientId)
+        {
+            string identityEndpoint = EnvironmentVariables.IdentityEndpoint;
+            string identityHeader = EnvironmentVariables.IdentityHeader;
+            string identityServerThumbprint = EnvironmentVariables.IdentityServerThumbprint;
+
+            if (string.IsNullOrEmpty(identityEndpoint) || string.IsNullOrEmpty(identityHeader) || string.IsNullOrEmpty(identityServerThumbprint))
+            {
+                return default;
+            }
+
+            Uri endpointUri;
+            try
+            {
+                endpointUri = new Uri(identityEndpoint);
+            }
+            catch (FormatException ex)
+            {
+                throw new AuthenticationFailedException(IdentityEndpointInvalidUriError, ex);
+            }
+
+            var httpHandler = new HttpClientHandler();
+
+            httpHandler.ServerCertificateCustomValidationCallback = ValidateMsiServerCertificate;
+
+            var customSslHttpPipline = HttpPipelineBuilder.Build(new TokenCredentialOptions { Transport = new HttpClientTransport(httpHandler) });
+
+            var innerPipeline = new CredentialPipeline(pipeline.AuthorityHost, customSslHttpPipline, pipeline.Diagnostics);
+
+            return new ServiceFabricManagedIdentitySource(innerPipeline, endpointUri, identityHeader, clientId);
+        }
+
+        private ServiceFabricManagedIdentitySource(CredentialPipeline pipeline, Uri endpoint, string identityHeaderValue, string clientId) : base(pipeline)
+        {
+            _endpoint = endpoint;
+            _identityHeaderValue = identityHeaderValue;
+            _clientId = clientId;
+        }
+
+        protected override Request CreateRequest(string[] scopes)
+        {
+            // covert the scopes to a resource string
+            string resource = ScopeUtilities.ScopesToResource(scopes);
+
+            Request request = Pipeline.HttpPipeline.CreateRequest();
+
+            request.Method = RequestMethod.Get;
+            request.Headers.Add("X-IDENTITY-HEADER", _identityHeaderValue);
```

Comment:
Service Fabric uses a "Secret" header: [docs](https://docs.microsoft.com/en-us/azure/service-fabric/how-to-managed-identity-service-fabric-app-code#acquiring-an-access-token-using-rest-api)
```suggestion
            request.Headers.Add("Secret", _identityHeaderValue);
```